hacking/release-procedure.rst

```diff
@@ -0,0 +1,350 @@
+
+
+.. _first_release:
+
+To make beta1 for the major release
+-----------------------------------
+
+1. Create stable-2.X and push
+2. Change version in stable-2.X:
+
+    a. Change devel to 2.X in :file:`docs/docsite/_themes/srtd/layout.html`
+
+3. Bump the **devel** version to 2.(X+1)
+
+    a. :file:`VERSION`
+    b. :file:`lib/ansible/release.py`
+    c. :file:`packaging/release/vars/versions.yml`
+
+        i. Determine a codename for 2.(X+1) (Led Zeppelin song name)
+
+4. Run ``ansible-playbook packaging/release/release.yml``
+
+    a. Enter the release branch: stable-2.X
+    b. Enter the release version: 2.X.0.0
+    c. Enter the release string (ie. 0.1.beta1, or just 1 for final releases): 0.1.beta1
+    d. Does this branch have git submodules? [no]: no
+    e. Is this a final release (not a beta/rc)? [no]: no
+    f. Push repositories upstream when done? [no]: no
+
+5. The release has been built into: :file:`packaging/release/ansible_release`  (as long as you use
+   the local connection on localhost) cd into that directory to do the rest of the work
+6. ``git log -p`` to sanity check
+
+    a. Check :file:`VERSION`
+    b. :file:`lib/ansible/release.py`
+
+7. ``git push``
+8. Wait for shippable to run on the stable-2.X branch and check that it is good
+
+    a. https://app.shippable.com/github/ansible/ansible/dashboard
+
+9. ``git push --tags``
+10. Push sha and tarfiles out (pre-release candidates do this manually, final releases do this in
+    jenkins)::
+
+        export AUSERNAME=tkuratomi
+        scp -P 5150 packaging/release/ansible_release/dist/* $AUSERNAME@74.207.229.77:
+        ssh -p 5150 $AUSERNAME@74.207.229.77
+        chmod 0644 ansible-*.tar.gz*
+        sudo mv ~/ansible-*.tar.gz* /var/www/html/releases/ansible/
+        sudo chown root:root /var/www/html/releases/ansible/ansible-*.tar.gz*
+
+11. http://jenkins.testing.ansible.com/job/Sync_release_mirrors/
+
+    a. Build now
+
+12. Start the public RPM package build:
+
+    a. http://jenkins.testing.ansible.com/job/Build_Ansible_Public_RPM_Branch/build?delay=0sec
+    b. For the branch use the release tag created above.
+    c. For the publish option use ``preview`` for an RC and ``release`` for a release.
+
+13. See if the package builds worked:
+
+    a. http://jenkins.testing.ansible.com/job/Build_Ansible_Public_RPM_Branch/
+
+14. Send email:
+
+    a. To ``ansible-devel``
+    b. `Template <https://gist.github.com/abadger/3171f11b769150ae931498facd85c80d>`_
+    c. Change the versions and the sha256sum.  Be sure to use https when copying links!
+    d. Get the email link from the `googlegroup archive
+       <https://groups.google.com/forum/#!forum/ansible-project>`_
+
+15. Non-release managers, please spread the rc announcement to:
+
+    a. Working Groups
+
+       i. In meetings
+       ii. On Agenda ticket
+
+    b. Network - Network team responsibility, just remind them
+       i. Network to code
+
```

Comment:
16. Distributed-CI (DCI) updated to test the new branch:

  a. Remind Network Team to update the DCI scripts to test the new branch

Reply:
@mattclay Do we have to configure anything to test new `stable-` branches, for example anything to enable code coverage jobs?

Reply:
Nothing is required for testing new stable branches. We only run nightly code coverage analysis on the devel branch.

Reply:
Reconfiguring the tests sounds like it is within someone else's abilities.    If there's specific people or channels that the RM should communicate the release on, we should say that specifically here (after item 14 and before item 15)